File: javascript/fib.js
function fib(n) {
    if (n <= 1) return n;
    return fib(n - 1) + fib(n - 2);
}

// 兼容 Node.js 和 QuickJS
const args = typeof process !== 'undefined' ? process.argv.slice(2) : scriptArgs.slice(1);

if (args.length === 0) {
    console.log('Usage: fib <n>');
    if (typeof process !== 'undefined') process.exit(1);
    else throw new Error('Usage: fib <n>');
}

const n = parseInt(args[0]);
const start = Date.now();
const result = fib(n);
const end = Date.now();

console.log(`JavaScript: fib(${n}) = ${result}`);
console.log(`Time: ${end - start}ms`);

File: go/fib.go
package main

import (
	"fmt"
	"os"
	"strconv"
	"time"
)

func fib(n int) int {
	if n <= 1 {
		return n
	}
	return fib(n-1) + fib(n-2)
}

func main() {
	if len(os.Args) < 2 {
		fmt.Println("Usage: fib <n>")
		os.Exit(1)
	}

	n, err := strconv.Atoi(os.Args[1])
	if err != nil {
		fmt.Println("Invalid number:", os.Args[1])
		os.Exit(1)
	}

	start := time.Now()
	result := fib(n)
	elapsed := time.Since(start)

	fmt.Printf("Go: fib(%d) = %d\n", n, result)
	fmt.Printf("Time: %dms\n", elapsed.Milliseconds())
}